src/spares/oc_addon_template.lsl

```diff
@@ -75,62 +80,83 @@ UserCommand(integer iNum, string sStr, key kID) {
         
     }
 }
-Link(string packet, integer iNum, string sStr, key kID){
-    string pkt = llList2Json(JSON_OBJECT, ["pkt_type", packet, "iNum", iNum, "addon_name", g_sAddon, "bridge", FALSE, "sMsg", sStr, "kID", kID]);
-    if(g_kCollar!= "" && g_kCollar!= NULL_KEY) 
+
+Link(string packet, integer iNum, string sStr, key kID, list lLinkParams){
+    list packet_data = ["pkt_type", packet, "iNum", iNum, "addon_name", g_sAddon, "bridge", FALSE, "sMsg", sStr, "kID", kID];
+    if (llGetListLength(lLinkParams) >= 2) // only append if there's a minimum of one key=value set
+    {
+        packet_data += lLinkParams;
+    }
+
+    string pkt = llList2Json(JSON_OBJECT, packet_data);
+    if (g_kCollar != "" && g_kCollar != NULL_KEY) {
         llRegionSayTo(g_kCollar, API_CHANNEL, pkt);
+    }
     else
+    {
         llRegionSay(API_CHANNEL, pkt);
+    }
 }
+
 key g_kCollar=NULL_KEY;
 integer g_iLMLastRecv;
+integer g_iLMLastSent;
+
 default
 {
     state_entry()
     {
         API_CHANNEL = ((integer)("0x"+llGetSubString((string)llGetOwner(),0,8)))+0xf6eb-0xd2;
         llListen(API_CHANNEL, "", "", "");
-        Link("online", 0, "", llGetOwner()); // This is the signal to initiate communication between the addon and the collar
+        Link("online", 0, "", llGetOwner(), ["optin", llDumpList2String([READY, SAY, SENSORDIALOG], "~")]); // This is the signal to initiate communication between the addon and the collar
```

Comment:
This is making assumptions that we want these to be received by an addon. Dialog is one of the most heavily used link signals. Best practice here would be to see my above comment and to initialize it as blank,  then give a commented out example of what it would look like if initialized.